src/scripts/components/layer/wv.layer.radio.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * NASA Worldview
+ *
+ * This code was originally developed at NASA/Goddard Space Flight Center for
+ * the Earth Science Data and Information System (ESDIS) project.
+ *
+ * Copyright (C) 2013 - 2017 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * Licensed under the NASA Open Source Agreement, Version 1.3
+ * http://opensource.gsfc.nasa.gov/nosa.php
+ */
+
+import React from 'react';
+import {Checkbox} from 'react-icheck';
+import renderHTML from 'react-render-html';
+
+/*
+ * A react component, Builds a toggle on/off layer selector
+ *
+ *
+ * @class LayerRadio
+ * @extends React.Component
+ */
+export default class LayerRadio extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      checked: this.props.enabled || false,
+      metadataIsVisible: this.props.expanded || false
+    };
+    this.handleDetails = this.handleDetails.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+  /*
+   * Toggle switch for the iCheck layer adder/remover
+   *
+   *
+   * @method handleChange
+   *
+   *
+   * @return {void}
+   */
+  handleChange() {
+    const newChecked = this.state.checked === false ? true : false;
+    this.setState({
+      checked: newChecked
+    });
+    if(newChecked) {
+      this.props.onState(this.props.layerId);
+    }
+    else {
+      this.props.offState(this.props.layerId);
+    }
+  }
+  /*
+   * Toggle switch for the metadata info button and close arrow
+   *
+   *
+   * @method handleDetails
+   *
+   * @param {e} event
+   *
+   * @return {void}
+   */
+  handleDetails(e) {
```

Comment:
I'm not sure what's going on with the `visible` state here. It seems to be redundant with `metadataIsVisible`, and it doesn't look like it's a part of the state when the component is initialized. Maybe a more descriptive name would help someone read the code.

The function name and description are hard to follow as well. It looks like the function maybe toggles some stuff related to the info button and close arrow, so maybe renaming it and rewriting it something like this would be helpful;

```
toggleMetadataButtons (e) {
  e.stopPropagation();
  var { metaDataIsVisible } = this.state;
  var { onChange, rowIndex, layerId } = this.props;
  this.setState({
    metaDataIsVisible: !metaDataIsVisible,
    visible: metaDataIsVisible?'hidden':'visible'
  });
  onChange(rowIndex);
  expand(layerId);
}
```

I'm not confident that I understand what this function is doing, so take this rewritten version with a grain of salt.